neurodamus/connection_manager.py

```diff
@@ -715,97 +714,99 @@ def target_gids(gids):
                 pathway_repr = "Pathway {} -> {}".format(src_target.name, dst_target.name)
             logging.info(" * %s. Created %d connections", pathway_repr, all_created)
 
-    def _get_conn_stats(self, dst_target):
-        """Estimates the number of synapses per type for the given destination target
-        Note: measurement is done without any restriction on the source target.
-        As so it can avoid counting the same synapse twice by keeping track of
-        target cells alone.
+    def _get_conn_stats(self, dst_target, src_target=None):
+        """Estimates the number of synapses for the given destination and source nodesets
 
         Args:
-            dst_target: The target to estimate synapses for
+            dst_nodeset: The target to estimate synapses for
+            src_nodeset: The source nodes allowed for the given synapses
 
         Returns:
-            A Counter object with the estimated synapses per type
+            The estimated number of synapses which would be created
 
-        Note:
-          - _src target is not considered so we count all inbound synapses
-          -  We will only consider gids which have not been accounted for yet.
         """
         BLOCK_BASE_SIZE = 5000
-        SAMPLES_PER_BLOCK = 100
+        SAMPLED_CELLS_PER_BLOCK = 100
 
-        # Get the raw gids for the destination target
-        raw_gids = dst_target.get_local_gids(raw_gids=True) if dst_target else self._raw_gids
+        # Get the raw gids for the destination target (in this rank)
+        local_gids = dst_target.get_local_gids(raw_gids=True) if dst_target else self._raw_gids
+        if not len(local_gids):  # Target is empty in this rank
```

Comment:
`if not local_gids:` should be equivalent, right?

Reply:
Numpy arrays are pickier than lists. Converting them to a boolean yields an error: `The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()`